### ISSUE TITLE
Another regression in JIT matching

### ISSUE THREAD

**Issue by zherczeg** [PhilipHazel/pcre2#12]
This is #2762 in the old Bugzilla, submitted by  Milian Wolff.

This is probably related to BUG 2621 except that I'm running with PCRE2 version 10.37 2021-05-26 and the specific issue from that bug doesn't reproduce anymore.

Instead, I'm running into the following reduced issue:

works:
```
printf '%s\n%s\n' '/\/([^\/]+)\/\d+/' '/A/B/0' | pcre2test 
PCRE2 version 10.37 2021-05-26
/\/([^\/]+)\/\d+/
/A/B/0
 0: /B/0
 1: B
```

does not work:
```
printf '%s\n%s\n' '/\/([^\/]+)\/\d+/' '/A/B/0' | pcre2test -jit
PCRE2 version 10.37 2021-05-26
/\/([^\/]+)\/\d+/
/A/B/0
No match
```

slight changes to the pattern make the issue go away

Also please note that bugzilla is missing an entry for version 10.37, as such I selected N/A for now.

**Comment by carenas** [comment]
Fixed in 10.38:

```
printf '%s\n%s\n' '/\/([^\/]+)\/\d+/' '/A/B/0' | pcre2test -jit
PCRE2 version 10.38 2021-10-01
/\/([^\/]+)\/\d+/
/A/B/0
 0: /B/0
 1: B
 ```

**Comment by PhilipHazel** [comment]
As this is fixed, I'm closing the issue.
